File: my_workspace/backend/app/schemas/draft.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class DraftCreate(BaseModel):
    form_type: str


class DraftRead(BaseModel):
    id: UUID
    user_id: UUID
    form_type: str
    form_data: dict
    current_step: int
    last_saved_at: datetime
    expires_at: datetime

    model_config = {"from_attributes": True}


class DraftUpdate(BaseModel):
    form_data: dict | None = None
    current_step: int | None = None

File: my_workspace/frontend/src/types/draft.ts
export interface DraftSession {
  id: string
  user_id: string
  form_type: string
  form_data: Record<string, unknown>
  current_step: number
  last_saved_at: string
  expires_at: string
}

export interface DraftUpdate {
  form_data?: Record<string, unknown> | object
  current_step?: number
}
